File: leetcode_notes/leetcode/0035.搜索插入位置/0035-搜索插入位置.java
class Solution {
    public int searchInsert(int[] nums, int target) {
        // Time O(logn) Space O(1)
        if(nums == null || nums.length == 0) return 0;
        
        int start = 0;
        int end = nums.length - 1;
        while(start + 1 < end) {
            int mid = (end - start) / 2 + start;
            if(target == nums[mid]) return mid;
            else if (target < nums[mid]) end = mid;
            else start = mid;
        }
        if(target <= nums[start]) {
            return start;
        } else if(target  <= nums[end]) {
            return end;
        } else {
            return end + 1;
        }
    }
}

File: leetcode_notes/leetcode_cn/0035.搜索插入位置/0035-搜索插入位置.py
class Solution:
    def searchInsert(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: int
        """
        
        try:
            nums.index(target)
            return(nums.index(target))
        except:
            if(target >= nums[-1]):
                return (len(nums))
            elif(target <= nums[0]):
                return(0)
            else:
                for i in range(1,len(nums)):
                    if (nums[i-1] <= target and nums[i] > target):
                        return (i)